src/Admin/Actions.php

```diff
@@ -55,14 +67,30 @@ public function save_admin_settings() {
 		unset( $post_data['action'] );
 		unset( $post_data['roadblock'] );
 
-		// Save Settings.
-		update_option( 'plausible_analytics_settings', $post_data );
+		if ( !empty( $post_data['domain_name'] ) && !empty( $post_data['custom_domain_prefix'] ) && !empty( $post_data['self_hosted_domain'] ) && !empty( $post_data['shared_link'] ) ) {
+			
+			$status = 'success';
+
+			update_option( 'plausible_analytics_settings', $post_data );
 
+			$message = esc_html__( 'Settings saved successfully.', 'plausible-analytics' );
+		} else {
+			$status = 'error';
+			$message = esc_html__( 'Something gone a wrong.', 'plausible-analytics' );
+		}
+		
+		// Save Settings.
+		$response = array(
+			'status'       => $status,
+			'message'      =>  $message,
+		);
 		// Send response.
 		wp_send_json_success(
 			[
-				'message' => esc_html__( 'Settings saved successfully.', 'plausible-analytics' ),
+				'message' => $message,
```

Comment:
@Ankur0812 Looks like you missed passing the `$response` array here which you created

Reply:
Not required, removed